### PR TITLE
Archive announcement

### LINKED ISSUE #968: Delete announcements in announcer tool
Archive instead of delete

### PR DESCRIPTION
**Pull request by avukkonen**
Closes #968

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/socialnavigation.js [hunk 1]
```diff
@@ -115,9 +115,9 @@
           }
           
         } else {
-          if (obj[value.name] == undefined)
+          if (obj[value.name] == undefined){
```
Comment (hkurhinen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/socialnavigation.js]: should this be if (typeof(obj[value.name]) === 'undefined') ?

File: muikku/src/main/webapp/resources/scripts/gui/announcer.js [hunk 2]
```diff
@@ -84,6 +90,64 @@
...
+      var archiveAnnouncements = function(values){
+
+         var ids = values.ids;
+         var arr = ids.split(',');
+        
+         var calls=$.map(arr, function(announcement){
+           return mApi().announcer.announcements.del(announcement);
+         });
+         
+         mApi({async: false}).batch(calls).callback(function(err, results){
```
Comment (hkurhinen) [inline muikku/src/main/webapp/resources/scripts/gui/announcer.js]: Probably not very big issue but using `async:false` in new code is not a good practice

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/AnnouncementController.java [hunk 3]
```diff
@@ -31,6 +31,10 @@ public Announcement create(
     );
   }
   
+  public List<Announcement> listUnArchived() {
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/AnnouncementController.java]: lowercase a

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/AnnouncementController.java [hunk 4]
```diff
@@ -38,4 +42,8 @@ public Announcement create(
   public Announcement findById(Long id) {
     return announcementDAO.findById(id);
   }
+  
+  public void archiveById(Long id) {
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/AnnouncementController.java]: archive(Announcement announcement)

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/dao/AnnouncementDAO.java [hunk 5]
```diff
@@ -24,8 +31,29 @@ public Announcement create(
     announcement.setCreated(created);
     announcement.setStartDate(startDate);
     announcement.setEndDate(endDate);
-
+    announcement.setArchived(false);
+    
     getEntityManager().persist(announcement);
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/dao/AnnouncementDAO.java]: return persist(announcement);

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/dao/AnnouncementDAO.java [hunk 6]
```diff
@@ -24,8 +31,29 @@ public Announcement create(
     announcement.setCreated(created);
     announcement.setStartDate(startDate);
     announcement.setEndDate(endDate);
-
+    announcement.setArchived(false);
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/dao/AnnouncementDAO.java]: Parameterize

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/dao/AnnouncementDAO.java [hunk 7]
```diff
@@ -24,8 +31,29 @@ public Announcement create(
...
+  
+  public void archiveById(Long id) {
+    Announcement announcement = findById(id);
+    if(announcement != null){
+      announcement.setArchived(true);
+      getEntityManager().persist(announcement);
+    }
+  }
+  
+  public List<Announcement> listUnArchived(){
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/dao/AnnouncementDAO.java]: listByArchived

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/rest/AnnouncementRESTModel.java [hunk 8]
```diff
@@ -53,4 +59,5 @@ public void setEndDate(Date endDate) {
   private Date created;
   private Date startDate;
   private Date endDate;
+  private boolean archived;
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/rest/AnnouncementRESTModel.java]: NotNull + nullable=false

Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/rest/AnnouncementRESTModel.java]: Object

File: muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/rest/AnnouncerRESTService.java [hunk 9]
```diff
@@ -87,4 +87,12 @@ private AnnouncementRESTModel createRESTModel(Announcement announcement) {
     restModel.setId(announcement.getId());
     return restModel;
   }
+
+  @DELETE
+  @Path("/announcements/{ID}")
```
Comment (anttileppa) [inline muikku-core-plugins/src/main/java/fi/muikku/plugins/announcer/rest/AnnouncerRESTService.java]: validation

**Pull request merged**